ldk/javascript/src/clipboard/index.ts

```diff
@@ -20,21 +20,60 @@ export interface Clipboard {
   /**
    * Starts listening to changes to the clipboard.
    *
-   * @param includeOliveHelpsEvents - if passed in true, callback will be called while olive helps window is in focus
+   * @param filterOptions - object, includesOliveHelps will callback will be called while olive helps window is in focus if true
    * @param callback - A function that's called whenever the clipboard's contents change.
    */
   listen(
-    includeOliveHelpsEvents: boolean,
+    filterOptions: FilterOptions,
     callback: (clipboardText: string) => void,
   ): Promise<Cancellable>;
 }
 
+type FilterOptions = {
+  includeOliveHelpsEvents: boolean;
+};
+
 export function listen(
-  includeOliveHelpsEvents: boolean,
+  filterOptions: FilterOptions,
   callback: (clipboardText: string) => void,
 ): Promise<Cancellable> {
-  oliveHelps.clipboard.includeOliveHelpsEvents(includeOliveHelpsEvents);
-  return promisifyListenable(callback, oliveHelps.clipboard.listen);
+  return new Promise((resolve, reject) => {
+    try {
+      let focusedWindow: string;
+
+      const setFocusedWindow = () =>
+        new Promise((res, rej) => {
+          oliveHelps.window.activeWindow((error, event) => {
+            if (error) {
+              return rej(error);
+            }
+            focusedWindow = event.title;
+
+            return res(focusedWindow);
+          });
+        });
+
+      oliveHelps.clipboard.listenAll(
+        async (error, event) => {
+          if (error) {
+            return error;
```

Comment:
I think we want to reject the error

Reply:
Do it not do that already? it is in a try catch statement that if there is an error then it reject(error). Line 74

Reply:
I wasnt sure if this error would be caught in the catch block or not since its a part of the inner callback but i think your right